File: packages/ts-sdk/test/unit/auth.test.ts
import { describe, expect, it } from 'vitest';
import { resolveAuthorizationHeader, StaticBearerTokenAuthProvider } from '../../src/treedx/index.js';

describe('auth helpers', () => {
  it('returns static bearer tokens', async () => {
    await expect(Promise.resolve(new StaticBearerTokenAuthProvider('token').getToken())).resolves.toBe('token');
  });

  it('formats authorization headers', async () => {
    await expect(resolveAuthorizationHeader({ token: 'abc' })).resolves.toEqual({ Authorization: 'Bearer abc' });
  });
});

File: packages/elixir-sdk/test/unit/auth_test.exs
defmodule TreeDxSdk.AuthTest do
  use ExUnit.Case, async: true

  test "static bearer auth resolves authorization header" do
    config = %TreeDxSdk.Config{token: "secret"}

    assert {:ok, {"Authorization", "Bearer secret"}} =
             TreeDxSdk.Auth.resolve_authorization_header(config)
  end
end
